client/src/components/VotingScreen/VotingScreen.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import { formatServerError } from '../../utils/errorFormatting';
+import './VotingScreen.css';
+
+class VotingScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: null,
+      voted: false,
+      selectedOption: '',
+    };
+    this.updateGameState = this.updateGameState.bind(this);
+    this.handleOptionChange = this.handleOptionChange.bind(this);
+    this.castVote = this.castVote.bind(this);
+  }
+
+  // Todo: Probably move to a helper since it's going to be used in other screens
+  async updateGameState() {
+    const { gameState, onGameStateChanged } = this.props;
+    const { groupName, currentPlayer } = gameState;
+    const { name: playerName } = currentPlayer;
+    try {
+      const response = await axios.get(`/api/get-game-status/${groupName}?playerName=${playerName}`);
+      onGameStateChanged(response.data);
+    } catch (error) {
+      this.setState({ error: formatServerError(error) });
+    }
+  }
+
+  handleOptionChange(voteEvent) {
+    const { voted } = this.state;
+    if (voted) {
+      return;
+    }
+    this.setState({
+      selectedOption: voteEvent.target.value,
+      error: '',
+    });
+  }
+
+  castVote() {
+    const { selectedOption } = this.state;
+    if (!selectedOption) {
+      this.setState({ error: 'select an option!   ' });
+      return;
+    }
+
+    this.setState({
+      voted: true,
```

Comment:
should we read this from the server response vs keeping individual state?

Reply:
absolutelay - i didnt do any servery things here. just wanted a skeleton front end. but yeah we shouldn't have redundant storage (frontend/backed) for hasVoted

we need a few more things in the gamestate (votes, prompts, drawings) but once we have those we can easily parse it in and not dupe any of that storage in the front end